### PR TITLE
[INF-16-39] feat: Add Sentry

### PR DESCRIPTION
**Pull request by phillip055**
Sentry will help us capture errors and increase visibility.

### CODE REVIEW

File: app.mjs [hunk 1]
```diff
@@ -29,6 +30,21 @@ const chainIdParamSchema = {
  * @param {import("fastify").FastifyInstance} fastify
  */
 export async function server(fastify, opts) {
+  Sentry.init({
+    dsn: process.env.SENTRY_DSN,
+    environment: process.env.NODE_ENV || "development"
```
Comment (numan) [inline app.mjs:35]: Can we add both of these to `.env.example` ?

Also document it in the README?

File: app.mjs [hunk 2]
```diff
@@ -29,6 +30,21 @@ const chainIdParamSchema = {
  * @param {import("fastify").FastifyInstance} fastify
  */
 export async function server(fastify, opts) {
+  Sentry.init({
```
Comment (numan) [inline app.mjs:33]: Also, can we make this optional? We don't necessarily want sentry running  in a dev environment

File: package.json [hunk 3]
```diff
@@ -62,6 +64,7 @@
     "eslint-config-prettier": "^8.3.0",
     "eslint-plugin-prettier": "^3.4.0",
     "prettier": "^2.3.2",
+    "sentry-testkit": "^3.3.7",
```
Comment (numan) [inline package.json:67]: Can we move this into dev dependencies and update the lock file?

**Review by numan — CHANGES_REQUESTED** [review]
Great work! Just added some minor feedback

**Comment by numan** [comment]
closes https://github.com/yearn/yearn-api/issues/32

**Comment by phillip055** [comment]
@numan made some requested changes, thanks for being patient :)

File: app.mjs (hunk 2 above)
Comment (numan) [inline app.mjs:33]: Can you please run this file through `prettier`? Just making sure we have consistent styling

File: app.mjs [hunk 4]
```diff
@@ -29,6 +30,21 @@ const chainIdParamSchema = {
...
+  fastify.setErrorHandler(function(err, request, reply) {
+    Sentry.withScope(scope => {
+        if (request.raw.ip) {
+          scope.setUser({
+            ip_address: request.raw.ip
+          });
+        };
+        scope.setTag("path", request.raw.url);
+        Sentry.captureException(err);
+    });
```
Comment (numan) [inline app.mjs:46]: Looks good. I think we just need to add a `reply.send(error)` here so it actually returns a 5xx response. As it is right now, it actually returns a 400 response.

This is going to throw off our monitoring tools

**Review by numan — CHANGES_REQUESTED** [review]
@phillip055 looks good. Just a couple more very minor things

**Comment by numan** [comment]
related to: https://github.com/yearn/yearn-api-infra/pull/2

**Comment by numan** [comment]
@phillip055 I'm actually going to merge this in and address my feedback in a followup PR

**Review by numan — APPROVED** [review]

**Pull request merged**